src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -263,50 +264,110 @@ void regclass_graph_Model(py::module m) {
-                :param partial_shapes: Index of Output.
-                :type partial_shapes: Dict[Output, PartialShape]
-                :return: void
+                :param partial_shape: New shape.
+                :type partial_shape: str
+                :return : void
              )");
 
+    function.def(
+        "reshape",
+        [](ov::Model& self, const py::dict& partial_shapes) {
+            std::map<ov::Output<ov::Node>, ov::PartialShape> new_shapes;
+            for (const auto& item : partial_shapes) {
+                std::pair<ov::Output<ov::Node>, ov::PartialShape> new_shape;
+                // check keys
+                if (py::isinstance<py::int_>(item.first)) {
+                    new_shape.first = self.input(item.first.cast<size_t>());
+                } else if (py::isinstance<py::str>(item.first)) {
+                    new_shape.first = self.input(item.first.cast<std::string>());
+                } else if (py::isinstance<ov::Output<ov::Node>>(item.first)) {
+                    new_shape.first = item.first.cast<ov::Output<ov::Node>>();
+                } else {
+                    throw py::type_error("Incorrect key type " + std::string(item.first.get_type().str()) +
+                                         " to reshape a model, expected keys as openvino.runtime.Output, int or str.");
+                }
+                // check values
+                if (py::isinstance<ov::PartialShape>(item.second)) {
+                    new_shape.second = item.second.cast<ov::PartialShape>();
+                } else if (py::isinstance<py::list>(item.second) || py::isinstance<py::tuple>(item.second)) {
+                    new_shape.second = Common::partial_shape_from_list(item.second.cast<py::list>());
+                } else if (py::isinstance<py::str>(item.second)) {
+                    new_shape.second = Common::partial_shape_from_str(item.second.cast<std::string>());
+                } else {
+                    throw py::type_error(
+                        "Incorrect value type " + std::string(item.second.get_type().str()) +
+                        " to reshape a model, expected values as openvino.runtime.PartialShape, str, list or tuple.");
+                }
+                new_shapes.insert(new_shape);
+            }
+            self.reshape(new_shapes);
+        },
+        py::arg("partial_shapes"),
+        R"( Reshape model inputs.
+
+            The allowed types of keys in the `partial_shapes` dictionary are:
+
+            (1) `int'
+            (2) `str`
+            (3) `openvino.runtime.Output`
+
+            The allowed types of values in the `partial_shapes` are:
+
+            (1) `openvino.runtime.PartialShape`
+            (2) `list`
+            (3) `tuple`
+            (4) `str`
+
+            To define a single dimension in the list use:
+
+            (1) `int`
+            (2) `[min, max]`
+            (3) `(min, max)`
+            (4) `openvino.runtime.Dimension`
+            (5) `str`
```

Comment:
This is not clear. Just listing `list`, `tuple`, `str` gives nothing info about the syntax of PS. Then we are talking about `a single dimension`, it is not clear how that are connected. Please add some words like "PartialShape is a list of dimension that can be represented as list, tuple and string with the following syntax" or something similar. Am I right that this will be used for auto-generated docs for Python API? If so, this is not enough.

Reply:
done